e2e/suites/management/list_operations_test.go

```diff
@@ -56,87 +55,147 @@ func TestListOperations(t *testing.T) {
-				assert.True(t, operationExists)
+				// Assert that every health controller operation in the final stage took some action
+				if pendingOp.DefinitionName == healthcontroller.OperationName {
+					assertHealthControllerTookAction(t, redisClient, scheduler, pendingOp)
+				}
 
-				assert.Equal(t, apiOperation.SchedulerName, scheduler.Name)
+				assert.Equal(t, finishedStatus, pendingOp.Status)
 			}
 
-			for _, apiOperation := range listOperationsResponse.FinishedOperations {
-				assert.NotEmpty(t, apiOperation.Id)
-				assert.Equal(t, apiOperation.Status, statusFinishedString)
+			err = managementApiClient.Do("GET", fmt.Sprintf("/schedulers/%s/operations?stage=final&page=2&perPage=%d", scheduler.Name, perPage), listOperationsRequest, listFinalOperationsResponse)
+			require.NoError(t, err)
+			assertPaginationInfo(t, listFinalOperationsResponse, uint32(10), uint32(2), uint32(5))
+			assert.Len(t, listFinalOperationsResponse.Operations, int(perPage)-1)
+			for _, pendingOp := range listFinalOperationsResponse.Operations {
+				if pendingOp.DefinitionName == test_operation.OperationName {
+					testOpsResponseCount++
+				}
 
-				_, operationExists := providers.ProvideDefinitionConstructors()[apiOperation.DefinitionName]
-				assert.True(t, operationExists)
+				// Assert that every health controller operation in the final stage took some action
+				if pendingOp.DefinitionName == healthcontroller.OperationName {
+					assertHealthControllerTookAction(t, redisClient, scheduler, pendingOp)
+				}
 
-				assert.Equal(t, apiOperation.SchedulerName, scheduler.Name)
+				assert.Equal(t, finishedStatus, pendingOp.Status)
 			}
+
+			err = managementApiClient.Do("GET", fmt.Sprintf("/schedulers/%s/operations?stage=final&page=3&perPage=%d", scheduler.Name, perPage), listOperationsRequest, listFinalOperationsResponse)
+			require.NoError(t, err)
+			assertPaginationInfo(t, listFinalOperationsResponse, uint32(10), uint32(3), uint32(5))
+			assert.Empty(t, listFinalOperationsResponse.Operations)
+
+			assert.Equal(t, numberOfFinishedTestOps, testOpsResponseCount)
 		})
 	})
 }
+
+func assertHealthControllerTookAction(t *testing.T, redisClient *redis.Client, scheduler *maestroApiV1.Scheduler, pendingOp *maestroApiV1.ListOperationItem) {
+	healthControllerDef := healthcontroller.SchedulerHealthControllerDefinition{}
+	definitionContents, err := redisClient.HGet(context.Background(), fmt.Sprintf("operations:%s:%s", scheduler.Name, pendingOp.Id), "definitionContents").Result()
+	require.NoError(t, err)
+	err = healthControllerDef.Unmarshal([]byte(definitionContents))
+	require.NoError(t, err)
+	assert.Equal(t, true, healthControllerDef.TookAction)
+}
+
+func assertPaginationInfo(t *testing.T, response *maestroApiV1.ListOperationsResponse, total uint32, page uint32, pageSize uint32) {
+	assert.GreaterOrEqual(t, *response.Total, total)
+	assert.GreaterOrEqual(t, *response.PageSize, pageSize)
```

Comment:
What do you think about segmenting the test by the operation status, when it is pending or active it would be `GreaterOrEqual` and when it is finished it must be `Equal`